src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1844,32 +1844,33 @@ internal static void AddDiscriminatorMethod(CodeClass newClass, string discrimin
     }
     private void CreatePropertiesForModelClass(OpenApiUrlTreeNode currentNode, OpenApiSchema schema, CodeNamespace ns, CodeClass model)
     {
-        if (schema?.Properties?.Any() ?? false)
-        {
-            model.AddProperty(schema
-                                .Properties
-                                .Select(x =>
-                                {
-                                    var propertySchema = x.Value;
-                                    var className = propertySchema.GetSchemaName().CleanupSymbolName();
-                                    if (string.IsNullOrEmpty(className))
-                                        className = $"{model.Name}_{x.Key.CleanupSymbolName()}";
-                                    var shortestNamespaceName = GetModelsNamespaceNameFromReferenceId(propertySchema.Reference?.Id);
-                                    var targetNamespace = string.IsNullOrEmpty(shortestNamespaceName) ? ns :
-                                                        rootNamespace?.FindOrAddNamespace(shortestNamespaceName) ?? ns;
-                                    var definition = CreateModelDeclarations(currentNode, propertySchema, default, targetNamespace, string.Empty, typeNameForInlineSchema: className);
-                                    if (definition == null)
+        if (!model.Properties.Any(static x => x.IsOfKind(CodePropertyKind.Custom))) // this redundant check on model properties is here to avoid race conditions
```

Comment:
Out of curiosity, what race conditions/situations are have you seen here?

Reply:
Sometimes the model gets processed multiple times because it's being created from different threads started from different paths. When that happens, properties get added multiple times.